src/handler/sign_psbt.c

```diff
@@ -1918,68 +1918,70 @@ sign_sighash_schnorr_and_yield(dispatcher_context_t *dc,
 
     bool error = false;
     cx_ecfp_private_key_t private_key = {0};
-    BEGIN_TRY {
-        TRY {
-            do {  // block executed once, only to allow safely breaking out on error
 
-                uint8_t *seckey =
-                    private_key.d;  // convenience alias (entirely within the private_key struct)
+    // IMPORTANT: Since we do not use any syscall that might throw an exception, it is safe to avoid
+    // using the TRY/CATCH block to ensure zeroing sensitive data.
 
-                uint32_t sign_path[MAX_BIP32_PATH_STEPS];
+    do {  // block executed once, only to allow safely breaking out on error
```

Comment:
If you removed the try catch you can also remove the `do .. while()` I guess :)

Reply:
there are still a couple of places where I use break out of the loop (and I can't use `return` because I have to make sure I call `explicit_bzero` below).